### DISCUSSION TITLE
Encoders work in reverse

### DISCUSSION THREAD

**Discussion by pbugdol** [FundamentalFrequency/.github#55] [Q&A]
I recently put the whole thing together and strangely enough the encoders are working in the opposite direction. Do you have any ideas on how to remedy this?

**Comment by pbugdol** [comment]
I managed to solve the problem by swapping the encoder pins in the Teensy software.

**Reply by mranisavljevic** [reply]
Do you have more details on exactly how to do this?

**Reply by mranisavljevic** [reply]
I found the answer for myself, thanks to this clue.  For anybody in the future who may experience this issue here's how to fix it.

Open the LMN-3-Firmware, provided in this repo. Navigate to src>main.cpp.  There you will find the encoder mapping for all 4 encoders right near the top of the file.  In each one you will see a pair of numbers representing the pins (and labeled as such).  Simply reverse the order of those numbers, save the file, and build & upload the firmware to the Teensy device again.

For example, the first encoder should look like this after completing the process:

```
CCRotaryEncoder enc1 = {
    {6, 5}, // pins
    {ENCODER_1},         // MIDI address (CC number + optional channel)
    1,      // optional multiplier if the control isn't fast enough
};
```
